src/tests/fuzzing/fuzz_piv_tool.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * fuzz_piv_tool.c: Fuzz target for piv-tool
+ *
+ * Copyright (C) 2022 Red Hat, Inc.
+ *
+ * Author: Veronika Hanulikova <vhanulik@redhat.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "libopensc/internal.h"
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include "fuzzer_reader.h"
+#include "fuzzer_tool.h"
+#undef stderr
+#define stderr stdout
+
+/* Rename main for calling in fuzz target */
+#define main _main
+/* Connect to virtual reader instead of real card*/
+#define util_connect_card(ctx, card, id, do_wait, verbose) fuzz_util_connect_card(ctx, card)
+# include "tools/piv-tool.c"
+#undef main
+
+static const uint8_t *reader_data = NULL;
+static size_t reader_data_size = 0;
+
+/* Use instead of util_connect_card() */
+int fuzz_util_connect_card(struct sc_context *ctx, struct sc_card **card)
+{
+	return fuzz_connect_card(ctx, card, NULL, reader_data, reader_data_size);
+}
+
+void initilize_global()
+{
+	/* Global variables need to be initialized
+	   since libFuzzer runs more test in one thread. */
+	reader_data = NULL;
+	reader_data_size = 0;
+	ctx = NULL;
+	card = NULL;
+	bp = NULL;
+	evpkey = NULL;
+	opt_reader = NULL;
+	opt_apdus = NULL;
+	opt_apdu_count = 0;
+
+	optind = 0;
+	opterr = 0; /* do not print out error messages */
+	optopt = 0;
+}
+
+void test_load(char *op, const uint8_t *data, size_t size)
+{
+	char *filename = NULL;
+	char *argv[] = {"./fuzz_piv", op, NULL /*ref*/, "-i", NULL /*filename*/, "-A", NULL /*admin*/, NULL};
+	int argc = 3;
```

Comment:
is this correct? Should not this be `6` to match the `argv` above?

Reply:
Thanks (`7` to be correct)